src/bindings/jna/libelektra/src/main/java/org/libelektra/Plugin.java

```diff
@@ -1,56 +1,83 @@
 package org.libelektra;
 
-import org.libelektra.exception.KDBException;
+import javax.annotation.Nonnull;
 
 /**
- * This is a Java representation of a plugin.
+ * Java representation of an Elektra plugin
  */
 public interface Plugin {
 
 	/**
-	 * Gets the config which was used to configure the plugin
-	 *
-	 * @return A KeySet containing the configuration of the plugin
+	 * Return value for plugin methods: An error occurred inside the plugin function
+	 */
+	static final int PLUGIN_STATUS_ERROR = -1;
+
+	/**
+	 * Return value for plugin methods: Everything went fine
+	 */
+	static final int PLUGIN_STATUS_SUCCESS = 1;
+
+	/**
+	 * Return value for plugin methods: Everything went fine and the function **did not** update the given key set / configuration
+	 */
+	static final int PLUGIN_STATUS_NO_UPDATE = 0;
+
+	/**
+	 * Return value for plugin methods: Everything went fine and a cache was hit
+	 */
+	static final int PLUGIN_STATUS_CACHE_HIT = 2;
```

Comment:
As this is only relevant for mmap plugins, we will probably not need it in Java. It will probably confuse people not knowing about mmap.

Reply:
removed